schema/schema.go

```diff
@@ -0,0 +1,416 @@
+// Package schema provides methods to build JSON schema of the collections
+// from user defined models.
+//
+// Collection model is declared using Go struct tags. The "tigrisdb" tag contains field properties.
+// The following properties are recognised:
+//  * primary_key - declares primary key of the collection
+//
+// Composite primary keys defined using optional tag indexes.
+//
+// Special tag value "-" can be specified to skip field persistence. Unexported fields are not persisted too.
+package schema
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path"
+	"path/filepath"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/tigrisdata/tigrisdb-client-go/driver"
+)
+
+const (
+	tagName       = "tigrisdb"
+	tagPrimaryKey = "primary_key"
+	tagPII        = "pii"
+	tagEncrypted  = "encrypted"
+	tagSkip       = "-"
+
+	typeInteger = "integer"
```

Comment:
have a file data_types.go and move these types there?

Reply:
I think, having many files with just few lines, only complicates the project navigation.

Reply:
that will help users navigate the supported data types by Tigris. If not moving it in a different file then move it in different const blocks so that if someone needs they can directly reference that code block.